### PR TITLE
chore: add and apply naming convention

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
So, with this, the type aliases will need to be a bit more consistent and concise.

---

No `I` prefixes for interfaces, since it's redundant.

Not allowed:
```ts
interface IInterface {}
```
Allowed:
```ts
interface Interface {}
```

No `Type` suffixes for exactly the same reason.

Not allowed:
```ts
interface TypeType  {}
```

Allowed:
```ts
interface Type {}
```

Finally, non-type related.

Not allowed:
```js
obj = { UPPER_CASE: property };
```

on grounds of weirdness.  They were only used by `react-hotkeys` and it seems to work just fine without them.

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
@ShaunSHamilton I took out some unused types.  Could you confirm if that's okay/not okay?

Also, https://github.com/freeCodeCamp/freeCodeCamp/pull/44010 covers some of the same ground and should be merged first so they don't have to deal with all the conflicts I will have created!

File: client/src/components/profile/components/certifications.tsx [hunk 1]
```diff
@@ -32,21 +32,21 @@ const mapStateToProps = (state: any, props: ICertificationProps) =>
     // @ts-ignore
   )(state, props);
 
-interface ICert {
+interface Cert {
```
Comment (ShaunSHamilton) [inline client/src/components/profile/components/certifications.tsx:35]: This is identical to `prop-types.ts > CurrentCert`

File: client/src/components/profile/profile.tsx [hunk 2]
```diff
@@ -4,13 +4,13 @@ import Helmet from 'react-helmet';
...
 import Timeline from './components/TimeLine';
+import Camper from './components/camper';
+import Certifications from './components/certifications';
+import HeatMap from './components/heat-map';
+import Portfolio from './components/portfolio';
 
-interface IProfileProps {
+interface ProfileProps {
   isSessionUser: boolean;
   user: {
```
Comment (ShaunSHamilton) [inline client/src/components/profile/profile.tsx:15]: Ideally, this should be typed with `prop-types.ts > User`

File: client/src/components/search/searchBar/search-bar.tsx [hunk 3]
```diff
@@ -48,14 +48,14 @@ type searchBarPropType = {
   isSearchFocused?: boolean;
   t?: (label: string) => string;
```
Comment (ShaunSHamilton) [inline client/src/components/search/searchBar/search-bar.tsx:49]: If you do end up making any of the above changes...

```suggestion
  t?: TFunction;
```

File: client/src/components/settings/privacy.tsx [hunk 4]
```diff
@@ -22,7 +22,7 @@ const mapStateToProps = createSelector(userSelector, user => ({
 const mapDispatchToProps = (dispatch: Dispatch) =>
   bindActionCreators({ submitProfileUI }, dispatch);
 
-type ProfileUIType = {
+type ProfileUI = {
```
Comment (ShaunSHamilton) [inline client/src/components/settings/privacy.tsx:25]: All of these should probably be added to the `User.profileUI` def in `prop-types.ts`

File: .eslintrc.json [hunk 5]
```diff
@@ -54,7 +54,46 @@
...
+            "selector": "typeLike",
+            "format": ["PascalCase"],
+            "custom": {
+              "regex": "^I[A-Z]|[a-zA-Z]Type$",
+              "match": false
+            }
+          },
+          {
+            "selector": "typeProperty",
+            "format": ["camelCase", "PascalCase"]
```
Comment (ShaunSHamilton) [inline .eslintrc.json:84]: Why do we allow `camelCase` for the type property?

Comment (ojeytonwilliams) [reply .eslintrc.json:84]: It handles properties of object types, so if we don't allow camel case, we can't write
```ts
type Type = {
  aProperty: string;
}
```

File: client/src/components/profile/components/certifications.tsx [hunk 6]
```diff
@@ -6,11 +6,12 @@ import { connect } from 'react-redux';
 import { createSelector } from 'reselect';
 
 import { certificatesByNameSelector } from '../../../redux';
+import type { CurrentCert } from '../../../redux/prop-types';
```
Comment (ShaunSHamilton) [inline client/src/components/profile/components/certifications.tsx:9]: Interesting that `type` is needed. Why is this?

File: client/src/components/profile/profile.tsx [hunk 7]
```diff
@@ -4,52 +4,16 @@ import Helmet from 'react-helmet';
...
-import Camper from './components/Camper';
-import Certifications from './components/Certifications';
-import HeatMap from './components/HeatMap';
-import Portfolio from './components/Portfolio';
+import { User } from './../../redux/prop-types';
 import Timeline from './components/TimeLine';
+import Camper from './components/camper';
+import Certifications from './components/certifications';
+import HeatMap from './components/heat-map';
+import PortfolioData from './components/portfolio';
```
Comment (ShaunSHamilton) [inline client/src/components/profile/profile.tsx:12]: Why the change of name?

Comment (ojeytonwilliams) [reply client/src/components/profile/profile.tsx:12]: :man_facepalming:

**Review by ShaunSHamilton — APPROVED** [review]
Thank you, for answering all my queries.

This GMTL 👍

**Review by naomi-lgbt — APPROVED** [review]
```ts
LGTM as Review;
```
